kernel/src/process/posix_thread/builder.rs

```diff
@@ -128,12 +143,21 @@ impl PosixThreadBuilder {
             sched_policy,
             fpu_context,
             is_init_process,
+            user_ns,
+            ns_proxy,
         } = self;
 
         let file_table = file_table.unwrap_or_else(|| RwArc::new(FileTable::new()));
 
         let fs = fs.unwrap_or_else(|| Arc::new(ThreadFsInfo::default()));
 
+        let user_ns = user_ns.unwrap_or_else(|| UserNamespace::get_init_singleton().clone());
+
+        let ns_proxy = ns_proxy.unwrap_or_else(|| {
+            assert!(Arc::ptr_eq(&user_ns, UserNamespace::get_init_singleton()));
```

Comment:
I'm not sure what you want to mean by writing this assertion. Do you want to mean `assert!(user_ns.is_none())` or something else?

Reply:
> Do you want to mean assert!(user_ns.is_none()) or something else?

Yes, I want to check the user passed `user_ns` is `None`. But in line 154, the `user_ns` has been set to initial user namespace. So we can only compare `user_ns` with initial user namespace here.

Reply:
This is confusing. How about moving this assertion to `pub fn ns_proxy(mut self, ...)`?

Reply:
I add such an assertion 
```rust
assert_eq!(user_ns.is_none(), ns_proxy.is_none());
let user_ns = user_ns.unwrap_or_else(|| UserNamespace::get_init_singleton().clone());
let ns_proxy = ns_proxy.unwrap_or_else(|| NsProxy::get_init_singleton().clone());
```